glitch/analysis/security.py

```diff
@@ -170,3 +180,66 @@ def check_condition(self, c: ConditionStatement, file: str) -> list[Error]:
             return errors + [Error('sec_no_default_switch', c, file, repr(c))]
 
         return errors
+
+    def check_unitblock(self, u: UnitBlock) -> list[Error]:
+        errors = super().check_unitblock(u)
+
+        missing_integrity_checks = {}
+        for au in u.atomic_units:
+            result = self.check_integrity_check(au, u.path)
+            if result:
+                missing_integrity_checks[result[0]] = result[1]
+                continue
+
+            if file := SecurityVisitor.check_has_checksum(au):
+                if file in missing_integrity_checks:
+                    del missing_integrity_checks[file]
+
+        errors += missing_integrity_checks.values()
+
+        if self.tech == Tech.docker and 'Dockerfile' not in u.name:
+            image = u.name.split(":")
+            if image[0] not in SecurityVisitor.__OFFICIAL_IMAGES:
+                errors.append(Error('sec_non_official_image', u, u.path, repr(u)))
```

Comment:
I think this would be better implemented with a [Smell Checker](https://github.com/sr-lab/GLITCH/wiki/5.-Analyses#smell-checkers---adding-specific-behavior). In this case, we would have a default one that does nothing and the one for Docker that does this check. It is more verbose, but I think it highlights better the distinctions. What do you think?

Reply:
Yes, I think it is a good idea, already pushed the change